### ISSUE TITLE
[OpenMP] 'omp unroll partial' loop generates too many iterations

### ISSUE THREAD

**Issue by mikerice1969** [llvm/llvm-project#55236]
In the following test the loop executes one more time than expected:

```c
void unroll_partial_remainder(int *A)
{
    #pragma omp parallel for
    #pragma omp unroll partial(4)
    for (int i = 0; i < 12; ++i)
        A[i] = i;
}

#include <stdio.h>
#include <assert.h>

int main() {

  int A[24];
  for(int i = 0; i<24; i++){ A[i]=0; }

  // write 12 elements 0 thru 11.
  unroll_partial_remainder(A);

  // Off by one?
  assert(A[12] == 0);

  return 0;
}
```

```console
$ clang -fopenmp -fopenmp-version=51 u.c && ./a.out
a.out: u.c:21: int main(): Assertion `A[12] == 0' failed.
Aborted (core dumped)

$ clang --version
clang version 15.0.0 (git@github.com:llvm/llvm-project ef87865b98fa25af1d2c045bab1268b2a1503374)
Target: x86_64-unknown-linux-gnu
```

**Comment by Meinersbur** [comment]
Thanks for the report. I fixed it in acec07005e038ab2891f235ae60ba2f0236bb952  after verifying a variety of other combinations are working as expected. I already did that at the patch stage, I don't know how this slipped through.

/cherry-pick acec07005e038ab2891f235ae60ba2f0236bb952

**Comment by tstellar** [comment]
Merged: 2f0a69c32a4c48473c187179a594ac035b244e74
